builtin/credential/aws/path_login.go

```diff
@@ -1047,18 +1047,23 @@ func (b *backend) pathLoginRenewIam(ctx context.Context, req *logical.Request, d
 		case !roleEntry.ResolveAWSUniqueIDs && strutil.StrListContains(roleEntry.BoundIamPrincipalARNs, canonicalArn): // check 2 passed
 		default:
 			// check 3 is a bit more complex, so we do it last
+			// only try to look up full ARNs if there's a wildcard ARN in BoundIamPrincipalIDs.
+			if !hasWildcardBind(roleEntry.BoundIamPrincipalARNs) {
+				return nil, fmt.Errorf("role %q no longer bound to ARN %q", roleName, canonicalArn)
```

Comment:
@kalafut Is it okay to return the role name as part of the error? This is based off the note above on information leakage, so wondering if where's trying to avoid this explicitly.

Reply:
Discussed with this with a few parties internally and we're good with this change. Returning the role, though it's an additional bit of information, seems to be okay since a valid token with additional permissions is required to query information about the role.